x-pack/plugins/synthetics/server/routes/monitor_cruds/add_monitor_project.ts

```diff
@@ -84,3 +90,29 @@ export const REQUEST_TOO_LARGE = i18n.translate('xpack.synthetics.server.project
   defaultMessage:
     'Delete request payload is too large. Please send a max of 250 monitors to delete per request',
 });
+
+export const validatePermissions = async (
+  { server, response, request }: RouteContext,
+  projectMonitors: ProjectMonitor[]
+) => {
+  const hasPublicLocations = projectMonitors.some(({ locations }) => (locations ?? []).length > 0);
+  if (!hasPublicLocations) {
+    return;
+  }
+
+  const elasticManagedLocationsEnabled =
+    Boolean(
+      (await server.coreStart?.capabilities.resolveCapabilities(request)).uptime
+        .elasticManagedLocationsEnabled
+    ) ?? true;
+  if (!elasticManagedLocationsEnabled) {
+    return ELASTIC_MANAGED_LOCATIONS_DISABLED;
+  }
+};
```

Comment:
```suggestion
const hasPublicLocations = (projectMonitors: ProjectMonitor[]) =>
  projectMonitors.some(({ locations }) => (locations ?? []).length > 0);

export const validatePermissions = async (
  { server, response, request }: RouteContext,
  projectMonitors: ProjectMonitor[]
) => {
  if (!hasPublicLocations(projectMonitors)) {
    return;
  }

  const elasticManagedLocationsEnabled =
    Boolean(
      (await server.coreStart?.capabilities.resolveCapabilities(request)).uptime
        .elasticManagedLocationsEnabled
    ) ?? true;
  if (!elasticManagedLocationsEnabled) {
    return ELASTIC_MANAGED_LOCATIONS_DISABLED;
  }
};
```

Nit, I think this is a bit cleaner since we don't reference the result elsewhere in the function.